### PR TITLE
telemetry: init envoy gateway telemetry apis

### PR DESCRIPTION
**Pull request by Xunzhuo**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Part of https://github.com/envoyproxy/gateway/issues/701

### CODE REVIEW

File: api/config/v1alpha1/envoygateway_types.go [hunk 1]
```diff
@@ -79,6 +84,20 @@ type EnvoyGatewaySpec struct {
 	ExtensionAPIs *ExtensionAPISettings `json:"extensionApis,omitempty"`
 }
 
+type EnvoyGatewayTelemetry struct {
+	// AccessLogs defines accesslog parameters for envoy gateway.
+	// If unspecified, will send default format to stdout.
+	// +optional
+	AccessLog *EnvoyGatewayAccessLog `json:"accessLog,omitempty"`
```
Comment (zirain) [inline api/config/v1alpha1/envoygateway_types.go:91]: what's accesslog for controlplane?

Comment (arkodg) [reply api/config/v1alpha1/envoygateway_types.go:91]: +1, dont think this is needed

File: api/config/v1alpha1/envoygateway_metric_types.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+// the root of the repo.
+
+package v1alpha1
+
+type EnvoyGatewayMetrics struct {
+	// Sinks defines the metric sinks where metrics are sent to.
+	Sinks []EnvoyGatewayMetricSink `json:"sinks,omitempty"`
+	// Matches defines configuration for selecting specific metrics instead of generating all metrics stats
+	// that are enabled by default. This helps reduce CPU and memory overhead in Envoy Gateway.
+	Matches []Match `json:"matches,omitempty"`
```
Comment (zirain) [inline api/config/v1alpha1/envoygateway_metric_types.go:13]: how will these work?
how to support Prometheus? EG should always support prometheus first.

**Comment by zirain** [comment]
can you spilit tracing support to a seperate PR?
IMHO, tracing support is not urgent, we can leave it for the furture.

**Comment by Xunzhuo** [comment]
Close it and resend one after design docs.

**Pull request closed without merging**